src/kagglehub/clients.py

```diff
@@ -120,7 +121,7 @@ def post(self, path: str, data: dict) -> dict:
             self._check_for_version_update(response)
             return response_dict
 
-    def download_file(self, path: str, out_file: str) -> None:
+    def download_file(self, path: str, out_file: str, resource_handle: Optional[ResourceHandle] = None) -> None:
```

Comment:
not sure if this is on purpose, but you didnt change how download_file is getting called anywhere in the code. is the change for the future if needed?
download is called in `http_resolver.py` and `test_kaggle_api_client`

Reply:
Good catch, I actually forgot to commit my latest change 🤦

Reply:
Can you also review the refactor PR please: https://github.com/Kaggle/kagglehub/pull/79. Thank you

Reply:
i commented there as well